pkg/snowflake/grant.go

```diff
@@ -138,7 +138,13 @@ func (gb *CurrentGrantBuilder) Share(n string) GrantExecutable {
 
 // Grant returns the SQL that will grant privileges on the grant to the grantee
 func (ge *CurrentGrantExecutable) Grant(p string) string {
-	return fmt.Sprintf(`GRANT %v ON %v %v TO %v "%v"`,
+	var template string
+	if p == `OWNERSHIP` {
+		template = `GRANT %v ON %v %v TO %v "%v" COPY CURRENT GRANTS`
+	} else {
+		template = `GRANT %v ON %v %v TO %v "%v"`
+	}
+	return fmt.Sprintf(template,
 		p, ge.grantType, ge.grantName, ge.granteeType, ge.granteeName)
 }
```

Comment:
Despite this fix, I've hit the `Dependent grant of privilege` error a few times. What I've found is in the [grant resource](https://registry.terraform.io/providers/chanzuckerberg/snowflake/latest/docs/resources/stage_grant), I had specified `privilege = "ownership"` (lowercase) which meant that it did not match `p == OWNERSHIP` (uppercase) and ran the query without the `copy current grants`.

Changing it to uppercase worked. So seems like it should be handled in this function as the provider doesn't throw an error in general if a lowercase value is provided and the docs don't mention it being case sensitive.